src/lint/track_config.nim

```diff
@@ -280,19 +302,19 @@ proc tidyJsonyErrorMsg(trackConfigContents: string): string =
   """.unindent()
   result = &"JSON parsing error:\nconfig.json{details}\n\n{bugNotice}"
 
-proc toTrackConfig(trackConfigContents: string): TrackConfig =
+proc init(T: typedesc[TrackConfig]; trackConfigContents: string): T =
   ## Deserializes `trackConfigContents` using `jsony` to a `TrackConfig` object.
   try:
     result = fromJson(trackConfigContents, TrackConfig)
   except jsony.JsonError:
     let msg = tidyJsonyErrorMsg(trackConfigContents)
     showError(msg)
 
-func getConceptSlugs(trackConfig: TrackConfig): HashSet[string] =
+func getConceptSlugs(concepts: Concepts): HashSet[string] =
   ## Returns a set of every `slug` in the top-level `concepts` array of a track
   ## `config.json` file.
   result = initHashSet[string]()
-  for con in trackConfig.concepts:
+  for con in concepts:
```

Comment:
@ErikSchierboom: FYI, in Nim 1.6 we can import `std/sugar` and instead write (with zero overhead!):

```Nim
collect(for con in concepts: {con.slug})
```

or

```Nim
collect:
  for con in concepts:
    {con.slug}
```


So after the Nim 1.6 release I'll probably write this kind of proc as a `collect` one-liner.

---

In Nim 1.4 we have to be more verbose:

```Nim
collect(initHashSet, for con in concepts: {con.slug})
```

or

```Nim
collect(initHashSet):
  for con in concepts:
    {con.slug}
```

---

See also:
- [Nim `collect` compared with Python list comprehensions](https://github.com/nim-lang/Nim/wiki/Nim-for-Python-Programmers#List-Comprehensions)
- `std/sugar` [docs](https://nim-lang.github.io/Nim/sugar.html#collect.m%2Cuntyped)
- [Nim 1.2.0 release notes](https://nim-lang.org/blog/2020/04/03/version-120-released.html)